source/class/qx/tool/compiler/ClassFile.js

```diff
@@ -1648,6 +1656,60 @@ qx.Class.define("qx.tool.compiler.ClassFile", {
               }
             }
 
+            // Are we looing at the Identifier `require`, and is it a
+            // function call (identified by having
+            // `path.node.arguments`? (A better implementation might
+            // look for a type of `CallExpression` at the top level of
+            // the node...) If so, we'll add
+            // qx-commonjs-browserified.js as an External so it gets
+            // pulled in automagically, but only if application-types
+            // is "browser" (and nothing else)
+            //
```

Comment:
That's a rational concern (esp given all the warnings in babel about not assuming context) but what you're doing here is correct IMHO.  Because you are in a CallExpression, you're just checking that the callee is an Identifier as opposed to a MemberExpression (or whatever).  So `require` is an Identifier but `someObject.require` is a MemberExpression which has two Identifiers within it.  

BTW I don't know if you've seen https://astexplorer.net/ ?  It's not 100% up to date with some ES6/Babel but it is fantastically useful in visualising the AST.